pkg/apis/numaflow/v1alpha1/vertex_types.go

```diff
@@ -343,13 +352,61 @@ func (v Vertex) GetPodSpec(req GetVertexPodSpecReq) (*corev1.PodSpec, error) {
 	}
 
 	if v.IsASource() && v.Spec.Source.Serving != nil {
-		servingContainer, err := v.getServingContainer(req)
+		servingSource := v.Spec.Source.Serving
```

Comment:
What is the reason of removing the abstracted function `getServingContainer()`?

Reply:
Till now, if Serving source is specified, we used to have a second container for starting Serving http server. Since we made Serving as a [builtin](https://github.com/numaproj/numaflow/pull/2301) [source](https://github.com/numaproj/numaflow/pull/2311), this second container is not needed.
Instead, we set the env variables to enable callback if Serving is used as a source.